horreum-backend/src/test/java/io/hyperfoil/tools/horreum/svc/AlertingServiceTest.java

```diff
@@ -776,6 +765,38 @@ public void testFindLastDatapoints(TestInfo info) throws IOException {
               .then().statusCode(200).extract().body().as(new ParameterizedTypeImpl(List.class, AlertingService.DatapointLastTimestamp.class));
    }
 
+   @org.junit.jupiter.api.Test
+   public void testUpdateVariablesHandlesNegativeId(TestInfo info) throws Exception {
+      Test test = createTest(createExampleTest(getTestName(info)));
+      Schema schema = createExampleSchema(info);
+      ChangeDetection cd = addChangeDetectionVariable(test, schema.id);
+
+      cd.config.put("filter", "min");
+      setTestVariables(test, "Value", new Label("value", schema.id), cd);//create
+
+      List<Variable> variables = variables(test.id);
+      assertEquals(1, variables.size());
+      assertNotNull(variables.get(0).id);
+
+      ChangeDetection problematicChangeDetection = new ChangeDetection();
+      problematicChangeDetection.id = -1;
+      problematicChangeDetection.model = RelativeDifferenceChangeDetectionModel.NAME;
+      problematicChangeDetection.config = JsonNodeFactory.instance.objectNode().put("threshold", 0.2).put("minPrevious", 2).put("window", 2).put("filter", "mean");
+      List<String> labels = Collections.singletonList("foobar");
+      Set<ChangeDetection> cdSet = Collections.singleton(problematicChangeDetection);
+
+      Variable throughput = new Variable();
+      throughput.id = -1;
+      throughput.testId = test.id;
+      throughput.name = "throughput";
+      throughput.labels = labels;
+      throughput.changeDetection = cdSet;
+
+      variables.add(throughput);
+
+      updateVariables(test.id, variables);//update
```

Comment:
can we add an assertion here as well to make sure the variables have been updated?

Reply:
sure